internal/pkg/client/cache/oras.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package cache
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/sylabs/singularity/internal/pkg/oras"
+	"github.com/sylabs/singularity/internal/pkg/sylog"
+)
+
+const (
+	// OrasDir is the directory inside the cache.Dir where oras images are cached
+	OrasDir = "oras"
+)
+
+// Oras returns the directory inside the cache.Dir() where oras images are cached
+func Oras() string {
+	return updateCacheSubdir(OrasDir)
+}
+
+// OrasImage creates a directory inside cache.Dir() with the name of the SHA sum of the image
+func OrasImage(sum, name string) string {
+	updateCacheSubdir(filepath.Join(OrasDir, sum))
```

Comment:
doesn't this return something like filepath.Join(cacheDir, OrasDir, sum)? If yes, maybe you can save that and use it in the next line as `return filepath.Jon(dir, name)`?

Reply:
That works ill change that